### ISSUE TITLE
Dockerizing qMRLab for Octave 

### ISSUE THREAD

**Issue by agahkarakuzu** [qMRLab/qMRLab#144]
I've already written an image processing friendly Octave Dockerfile. Putting a bit more effort would do the trick. 

Here is a perfect qMRLab on Docker themed illustration for motivation :D 

_The ocean is an umbrella by Igres Leandro_

![image](https://user-images.githubusercontent.com/9632322/33469552-82012408-d631-11e7-9c45-4a557eb17dcd.png)

**Comment by mathieuboudreau** [comment]
What's the name of your new branch to work on this feature?

**Comment by agahkarakuzu** [comment]
We don't need a specific branch for this. Once master is ready to roll out, we can dockerhub it.

**Comment by mathieuboudreau** [comment]
Sorry, I'm not sure to understand what you are proposing or what dockerhubing a master branch means?

**Comment by agahkarakuzu** [comment]
All we need to define on a dockerfile is the environment (octave and packages), github link to the qMRLab and few other adjustments to prepare a template. I don't think that we need to change anything in the project itself.

This is just to provide people with a lightweight container with qMRLabready to use in it. This is also going to be helpful for binder (online executable jupyter notebooks).

**Comment by mathieuboudreau** [comment]
Oh ok, I didn't know we were going to actively promote using qMRLab in Octave instead of Matlab (to my knowledge we've really only been using Octave because of Travis compatibility). Is this a feature that was requested by some users? I thought the setup of qMRLab on Matlab was pretty very quick painless, but if this helps with Jupyter notebooks that might be interesting.

A+

Mathieu

**Comment by agahkarakuzu** [comment]
@mathieuboudreau @jcohenadad I've pushed the first Docker image for qMRLab on DockerHub and edited neuropoly/qmrlab page to give necessary directives to users. 

https://hub.docker.com/r/neuropoly/qmrlab/

The image forks qMRLab from the `bids` branch (temporarily), since Dockerfile and example jupyter notebook are there. 

After merging `bids` into the `master`, I will create automated build, so that each merge into master will trigger a new image build on DockerHub.

**Comment by agahkarakuzu** [comment]
@mathieuboudreau No worries. We will have this working as long as we have bids branch, we can definitely take our time for this. 

I built this one on my local, it took approx 40 minutes. This is not a usual docker build, I need to say. It is quite heavy, because `Dockerfile` configures and installs Octave and some of its dependencies (otherwise it is not even showing 8bit images).  Resultant image is 8.1GB for now. When I'm more available, I will look into ways to make it lightweight (on Alpine or Debian instead of Ubuntu). On the other hand, MATLAB runtime (mcr) approach would solve this problem at its root.

**Comment by agahkarakuzu** [comment]
This issue has been resolved. Docker images of qMRLab will be available at [this Dockerhub organization](https://cloud.docker.com/orgs/qmrlab/teams).
